cmd/vicadmin/vicadm.go

```diff
@@ -140,6 +160,37 @@ func configureReaders() []entryReader {
 	return readers
 }
 
+type authenticator interface {
```

Comment:
Curious.  Why is the interface private to the package? Is there harm in making it public?  I'm wondering if this can be used elsewhere wholesale.  And if so, it should be in a `pkg/`.  Up to you.